helpers/class.TestSessionStorage.php

```diff
@@ -179,10 +176,11 @@ public function persist(AssessmentTestSession $assessmentTestSession)
     /**
      * @param AssessmentTestSession $session
      */
-    protected function lockSession(AssessmentTestSession $session)
+    public function lockSession(AssessmentTestSession $session)
     {
         $lock = $this->createLock('AssessmentTestSession_' . $session->getSessionId(), 30);
         $lock->acquire(true);
+        $session->setReadOnly(false);
```

Comment:
why session locking means that it become writeable?

Reply:
Good question. I think it was used to make it more clear from client point of view, you set access mode (readOnly) instead of saying lock/don't lock.

Reply:
It is clear, I'm wondering : why the method which named as lockSession can change state of the session, not the lock only, but also it can setReadOnly to false